src/components/shared/Button.tsx

```diff
@@ -0,0 +1,136 @@
+interface ButtonProps {
+  type?: string;
+  text?: string;
+  textColor?: string;
+  backColor?: string;
+  border?: string;
+  onClick?: () => void; // 버튼 클릭 시 실행할 함수
+  isActive?: boolean; // 버튼 or 탭 클릭 시 "UI 스타일만" 변경(배경 or text)
+  disable?: boolean; // 버튼 비활성화 버튼(비활성화+UI스타일 변경 - form 에 사용)
+}
+
+/* ============ 사용 예시 ============ */
+// type은 switch 문에 주석 달아놨으니 보고 사용하시면 됩니다
+// 똑같은 크기의 다른 역할(ex. 모달에서 취소/제출 버튼)은 textColor, backColor, border 지정해주셔야 합니다.
+// 버튼의 종류가 다양해서 놓친 게 있을 수 있으니 DM으로 말씀해주세여 :)
+{
+  /* <Button
+        type="start"
+        text="비회원으로 시작하기"
+        textColor="#ffffff"
+        backColor="#000000"
+        border="1px solid #000000"
+      />
+   */
+}
+const Button = ({
+  type,
+  text,
+  textColor,
+  backColor,
+  border,
+  onClick,
+  isActive,
+  disable = false,
+}: ButtonProps) => {
+  // 크기에 따라 Tailwind 클래스 설정
+  // 기본: 모바일, sm: 태블릿, md: PC
+  let sizeClass = '';
+  let activeClass = '';
+
```

Comment:
아까 리뷰를 남겼다 생각했는데 날라간것 같아서 다시 남겨봅니다! 컴포넌트 리랜더링시 불필요한 연산을 막기 위해 useMemo등을 사용하여 메모제이션하는 것을 권장드립니다! 혹시 다른 의도나 제가 잘못알고 있는 것이 있다면 답글 달아주시면 감사하겠습니다.
```suggestion
const sizeClass = useMemo(()=>{
  switch(type) {
  //...
  }},[type, isActive])

```

Reply:
리렌더링 작업은 리팩토링 하는 시간에 추가로 할 예정입니다! 
코멘트 남겨주셔서 감사합니다 :)